### ISSUE TITLE
dotnet store crashed with The process cannot access the file rs.assets.cache

### ISSUE THREAD

**Issue by pakrym** [dotnet/cli#9092]
## Steps to reproduce
1.
```
> cat D:\temp\store\rs.csproj

<Project Sdk="Microsoft.NET.Sdk"><ItemGroup>
<PackageReference Include="Microsoft.AspNetCore.AzureAppServices.HostingStartup" Version="2.1.0-preview2-final" />
<PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.1.0-preview2-final" />
</ItemGroup></Project>
```
2. 
`dotnet.exe store -m d:\temp\store\rs.csproj -f netcoreapp2.1 --framework-version 2.1.0-preview3-26413-05 -r win7-x64 --output d:\temp\store --skip-optimization`

## Expected  behavior
No error

## Actual behavior
```
icrosoft (R) Build Engine version 15.7.145.53551 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for d:\temp\store\rs.csproj...
  Restoring packages for d:\temp\store\rs.csproj...
  Generating MSBuild file C:\Users\pakrymet\AppData\Local\Temp\2yki0y3w.r4a\Microsoft.AspNetCore.Hosting_2.1.0-preview2-final\rs.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\pakrymet\AppData\Local\Temp\2yki0y3w.r4a\Microsoft.AspNetCore.Hosting_2.1.0-preview2-final\rs.csproj.nuget.g.targets.
  Restore completed in 338.28 ms for d:\temp\store\rs.csproj.
  Generating MSBuild file C:\Users\pakrymet\AppData\Local\Temp\2yki0y3w.r4a\Microsoft.AspNetCore.AzureAppServices.HostingStartup_2.1.0-preview2-final\rs.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\pakrymet\AppData\Local\Temp\2yki0y3w.r4a\Microsoft.AspNetCore.AzureAppServices.HostingStartup_2.1.0-preview2-final\rs.csproj.nuget.g.targets.
  Restore completed in 518.21 ms for d:\temp\store\rs.csproj.
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018: The "ResolvePackageAssets" task failed unexpectedly. [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018: System.IO.IOException: The process cannot access the file 'd:\temp\store\obj\Debug\netcoreapp2.1\win7-x64\rs.assets.cache' because it is being used by another process. [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheWriter..ctor(ResolvePackageAssets task, Stream stream) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheReader.CreateReaderFromDisk(ResolvePackageAssets task, Byte[] settingsHash) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheReader..ctor(ResolvePackageAssets task) [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ReadItemGroups() [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ExecuteCore() [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [d:\temp\store\rs.csproj]
C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(201,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [d:\temp\store\rs.csproj]
  Files were composed in d:\temp\store\x64\netcoreapp2.1\
  The list of packages stored is in d:\temp\store\x64\netcoreapp2.1\artifact.xml
```

## Environment data
`dotnet --info` output:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300-preview2-008533
 Commit:    996eb6c92c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\pakrymet\.dotnet\x64\sdk\2.1.300-preview2-008533\

Host (useful for support):
  Version: 2.1.0-preview3-26413-05
  Commit:  967b9806b2

```

Marking as blocking partner as this breaks ASP.NET Core site extension build.

**Comment by livarcocc** [comment]
Is this consistent? Are there other operations running besides the store command?

**Comment by livarcocc** [comment]
@nguerrera can you please investigate?

**Comment by pakrym** [comment]
> Is this consistent?
Happens in ~70% of invocations.

> Are there other operations running besides the store command?
No.

**Comment by nguerrera** [comment]
Fixed by dotnet/sdk#2159
